src/directConnect.ts

```diff
@@ -0,0 +1,54 @@
+import { ExtensionContext, ViewColumn } from "vscode";
+import { registerCommandWithLogging } from "./commands";
+import { WebviewPanelCache } from "./webview-cache";
+import connectionFormTemplate from "./webview/direct-connect-form.html";
+import { randomUUID } from "crypto";
+import { post } from "./webview/direct-connect-form";
+import { handleWebviewMessage } from "./webview/comms/comms";
+type MessageSender = OverloadUnion<typeof post>;
+type MessageResponse<MessageType extends string> = Awaited<
+  ReturnType<Extract<MessageSender, (type: MessageType, body: any) => any>>
+>;
+
+export const registerDirectConnectionCommand = (context: ExtensionContext) => {
+  const directConnectionCommand = registerCommandWithLogging("confluent.connections.direct", () =>
+    openDirectConnectionForm(),
+  );
+  context.subscriptions.push(directConnectionCommand);
+};
+
+const directConnectWebviewCache = new WebviewPanelCache();
+
+export function openDirectConnectionForm(): void {
+  // Set up the webview, checking for existing form for this topic
```

Comment:
Do we want any kind of caching here if it's an empty form? Or is this where we could pass `openDirectConnectionForm` a connection ID (UUID) and have it look up the connection details?
```suggestion
  // Set up the webview, checking for existing form for this connection
```

Reply:
yeah, as it is written now there is no real caching (new UUID each time the command is called) but I think it'll be useful when we add edit capabilities. I think we might also consider not allowing users to open more than one "new" connection form at a time.